### PR TITLE
Adding FinalProductProduct service and test

### PR DESCRIPTION
**Pull request by DianaMD21**
This branch contains the FinalProductProduct service and test.

### CODE REVIEW

File: test/util/fixtures/FinalProductProductFixtures.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.Optional;
+
+public class FinalProductProductFixtures {
+  public static FinalProductProduct buildFinalProductProduct(
+      FinalProductProduct finalProductProduct, FinalProductProduct finalProductProductExample) {
+    BaseEntityFixtures.buildBaseEntity(finalProductProduct, finalProductProductExample);
+    finalProductProduct.setFinalProduct(
+        Optional.ofNullable(finalProductProductExample)
+            .map(FinalProductProduct::getFinalProduct)
+            .orElse(null));
```
Comment (reynoldmorel) [inline test/util/fixtures/FinalProductProductFixtures.java:13]: Use final product fixture as default instead null

File: test/util/fixtures/FinalProductProductFixtures.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    finalProductProduct.setProduct(
+        Optional.ofNullable(finalProductProductExample)
+            .map(FinalProductProduct::getProduct)
+            .orElse(null));
```
Comment (reynoldmorel) [inline test/util/fixtures/FinalProductProductFixtures.java:17]: Use product fixture instead of null

**Review by reynoldmorel — CHANGES_REQUESTED** [review]
In general please use final product and product fixtures instead of using null or creating them on the fly

File: test/util/fixtures/FinalProductProductFixtures.java (hunk 1 above)
Comment (DianaMD21) [reply test/util/fixtures/FinalProductProductFixtures.java:13]: solved

File: test/util/fixtures/FinalProductProductFixtures.java (hunk 2 above)
Comment (DianaMD21) [reply test/util/fixtures/FinalProductProductFixtures.java:17]: solved

**Comment by DianaMD21** [comment]
> In general please use final product and product fixtures instead of using null or creating them on the fly

I added FinalProductFixtures, and since its not mandatory when creating a product setting the finalProductProduct list, i left it as null. The same thing happens with the finalProductProducts list in FinalProductFixtures, i also made the list null

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**